File: 프로그래머스/2/12913. 땅따먹기/땅따먹기.java
import java.util.*;

class Solution {
    int solution(int[][] land) {
        int answer = 0;
        int len = land.length;
        
        for(int i=1; i<len; ++i){
            land[i][0] += Math.max(Math.max(land[i-1][1], land[i-1][2]), land[i-1][3]);
            land[i][1] += Math.max(Math.max(land[i-1][0], land[i-1][2]), land[i-1][3]);
            land[i][2] += Math.max(Math.max(land[i-1][0], land[i-1][1]), land[i-1][3]);
            land[i][3] += Math.max(Math.max(land[i-1][0], land[i-1][1]), land[i-1][2]);
        }
        
        answer = Math.max(Math.max(Math.max(land[len-1][0], land[len-1][1]), land[len-1][2]), land[len-1][3]);

        return answer;
    }
}

File: 프로그래머스/2/12913. 땅따먹기/땅따먹기.py
def solution(land):
    answer = 0

    for i in range(1, len(land)):
        for j in range(4):
            maxNum = 0
            for k in range(4):
                if j==k: continue
                maxNum = max(maxNum, land[i-1][k])
                
            land[i][j] += maxNum
    
    answer = max(land[len(land)-1])
    
    return answer